atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java

```diff
@@ -177,14 +178,22 @@ public IndexDefinition getIndex(TableReference indexRef) {
      */
     public void validate() {
         // Try converting to metadata to see if any validation logic throws.
-        for (TableDefinition d : tableDefinitions.values()) {
-            d.toTableMetadata();
-            d.getConstraintMetadata();
+        for (Entry<String, TableDefinition> entry : tableDefinitions.entrySet()) {
+            try {
+                entry.getValue().validate();
+            } catch (Exception e) {
+                Throwables.rewrapAndThrowIfInstance("Failed to validate table " + entry.getKey(), e, RuntimeException.class);
```

Comment:
I really wanted to wrap the exception to add the table name since this is invaluable when debugging this issue (for example this is the only way I was able to fix internal code that began to fail).  Other options include (1) passing the table name to validate or (2) having validate the internal validates throw errors.